### PR TITLE
DataLoader: handling of non-scalar data

### PR DESCRIPTION
**Pull request by gracesong312**
Main updates to functionality:
1. WaveformTable (and general Table) loading
- Used to: manually explode columns of a WaveformTable so that one WaveformTable got loaded as three Arrays (for `dt`, `t0`, and `values`)
- Now: Load Tables as Tables, only explodes columns if the user requests a DataFrame, which it does in `get_dataframe` 
2. ArrayOfEqualSizedArray and VectorOfVector loading
AOESA will get loaded as VoV because the size of an array across multiple channels may be different. Added a constructor to VoV that builds a VoV out of a list of lists in order to accommodate this. 
3. Support `sortby` in FileDB config
Currently, FileDB sorts the dataframe by the key "timestamp," which is fine for L200, but test stands may not use a naming convention that includes "timestamp", so the user can now specify which key should be used for sorting. It still defaults to "timestamp" if no `sortby` is specified. 

For testing, I'm not entirely sure what's failing in the codecov and docs, I really don't think I changed anything. Also, I've tested VoV loading separately, but there isn't really good VoV data in the test suite, so I haven't added that in the tests folder.

### CODE REVIEW

File: src/pygama/flow/file_db.py [hunk 1]
```diff
@@ -192,6 +192,11 @@ def set_config(self, config: dict, config_path: str = None) -> None:
         self.tier_dirs = self.config["tier_dirs"]
         self.table_format = self.config["table_format"]
 
+        if "sortby" in self.config.keys():
+            self.sortby = self.config["sortby"]
+        else:
+            self.sortby = "timestamp"
+
```
Comment (gipert) [inline src/pygama/flow/file_db.py:199]: This can be one line using `dict.get(key, default_value)`

File: src/pygama/lgdo/vectorofvectors.py [hunk 2]
```diff
@@ -55,35 +59,48 @@ def __init__(
...
         """
-        if cumulative_length is None:
-            self.cumulative_length = Array(
-                shape=(shape_guess[0],), dtype="uint32", fill_val=0
+        if listoflists is not None:
+            cum_length = []
+            for vec in listoflists:
+                cum_length.append(len(vec))
+            self.flattened_data = Array(
+                nda=np.array([item for sublist in listoflists for item in sublist])
```
Comment (gipert) [inline src/pygama/lgdo/vectorofvectors.py:71]: Python for loops with built-in lists... Can this be made more efficient with Numpy?

**Review by gipert — CHANGES_REQUESTED** [review]
I noticed some functions without type hints, could you add them please?

File: src/pygama/lgdo/vectorofvectors.py (hunk 2 above)
Comment (gracesong312) [reply src/pygama/lgdo/vectorofvectors.py:71]: So the issue is that numpy doesn't like jagged arrays. I'm not sure what an efficient way around that would be.

Comment (jasondet) [reply src/pygama/lgdo/vectorofvectors.py]: I think I found a more efficient way. It should be good enough for us.

Comment (gipert) [reply src/pygama/lgdo/vectorofvectors.py]: In the long term we'll want to have more interoperability between `VectorOfVectors` and [`awkward.Array`](https://awkward-array.org/doc/main/index.html). Once #377 is addressed, we could support passing an awkward array as `listoflists`, that should be the fastest.

**Comment by gipert** [comment]
pre-commit.ci run

**Comment by gipert** [comment]
The pre-commit.ci failure seems an upstream problem to me, let's see if they fix it by today.

**Review by gipert — APPROVED** [review]

**Comment by jasondet** [comment]
@gipert can you check the error?

**Comment by jasondet** [comment]
I figured it out.

**Pull request merged**